src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java

```diff
@@ -949,4 +953,31 @@ private void processSearchComposite(final SearchCompositeDto<ObservationUnitsSea
 			searchDTO.setSearchRequest(searchRequest);
 		}
 	}
+
+	private void correctKSUDateFormatIfNecessary(final Table<String, String, String> table, final List<MeasurementVariable> measurementVariables) {
+
+		final List<MeasurementVariable> dateVariables = measurementVariables.stream().filter(measurementVariable -> measurementVariable.getDataTypeId()!=null && measurementVariable.getDataTypeId() == TermId.DATE_VARIABLE
+			.getId()).collect(Collectors.toList());
+		if (dateVariables != null) {
+			final List<String> dateMeasurementVariables = dateVariables.stream().map(measurementVariable -> measurementVariable.getName()).collect(Collectors.toList());
+			if (dateMeasurementVariables != null) {
+				for (final String obsUnit : table.rowKeySet()) {
+					for (final String colVariable : table.columnKeySet()) {
+						if (dateMeasurementVariables.contains(colVariable)) {
+							String value = table.get(obsUnit, colVariable);
+							if (Util.isDateMatchPattern(value, Util.DATE_AS_NUMBER_FORMAT_KSU)) {
+								final Date ksuParsed = Util.tryParseDateAccurately(value, Util.DATE_AS_NUMBER_FORMAT_KSU);
```

Comment:
Hi @gelli-leafnode , isDateMatchPattern is called twice. One in the IF condition and another inside tryParseDateAccurately(). Maybe rename tryParseDateAccurately to parseKSUFormatDate and just parse the date there without the checking if it's in KSU format.

Reply:
@mderamos , 
To avoid multiple call for method Util.isDateMatchPattern. Reverting logic to use tryParseDateAccurately. 
Thanks